python/dgl/ops/edge_softmax.py

```diff
@@ -102,8 +104,40 @@ def edge_softmax(graph, logits, eids=ALL, norm_by='dst'):
                 [0.5000],
                 [1.0000],
                 [0.5000]])
+
+
+    Examples on a heterogeneous graph
+    ---------------------------------
+
+    Create a heterogeneous graph and initialize its edge features.
+
+    >>> hg = dgl.heterograph({
+    ...     ('user', 'follows', 'user'): ([0, 0, 1], [0, 1, 2]),
+    ...     ('developer', 'develops', 'game'): ([0, 1], [0, 1])
+    ...     })
+    >>> edata_follows = th.ones(3, 1).float()
+    >>> edata_develops = th.ones(2, 1).float()
+    >>> edata_dict = {('user', 'follows', 'user'): edata_follows,
+    ... ('developer','develops', 'game'): edata_develops}
+
+    Apply edge softmax over hg normalized by source nodes:
+
+    >>> edge_softmax(hg, edata_dict, norm_by='src')
+        (tensor([[1.],
+        [1.]]), tensor([[0.5000],
+        [0.5000],
+        [1.0000]]))
```

Comment:
The heterogeneous APIs use tuple in the forward function (so that backward can keep tracks), also for storing intermediate and final outputs. One can explicitly convert the tuple to dict. Should it be a integrated part of the api?

Reply:
I think so.  This function is exposed to users so I would prefer explicitly converting it to a dict.  Otherwise users will need to figure out how to map it back (which they will almost always do).

Reply:
That's actually true! Converted back to dict in the new commit. Thanks.